### PR TITLE
Add tests for some of the assumptions of the test suite

### LINKED ISSUE #3827: Document assumptions and add tests for them
It would be beneficial to have a proper list of what the tests assume about the browsing environment to ensure people are running the tests correctly and getting correct results.

See, for example, https://github.com/w3c/csswg-test/issues/1097 when it comes to issues related to this.

The CSS test suite home pages have a list of assumptions, for example at http://test.csswg.org/suites/css21_dev/nightly-unstable/

Most of these are likely things we want to document and test for, so…
- [x] The X/HTML div element is assigned display: block; and no other property declaration.
- [x] The X/HTML span element is assigned display: inline; and no other property declaration.
- [x] The X/HTML p element is assigned display: block;
- [x] The X/HTML li element is assigned display: list-item;
- [x] The X/HTML table elements table, tbody, tr, and td are assigned the display values table, table-row-group, table-row, and table-cell, respectively.
- [ ] The device can display the sixteen color values associated with the color keywords black, white, gray, silver, red, green, blue, purple, yellow, orange, teal, fuchsia, maroon, navy, aqua, and lime as distinct colors.
- [ ] The UA is set to print background colors and, if it supports graphics, background images.
- [ ] The UA implements reasonable page-breaking behavior; e.g., it is assumed that UAs will not break at every opportunity, but only near the end of a page unless a page break is forced.
- [ ] The UA implements reasonable line-breaking behavior; e.g., it is assumed that spaces between alphanumeric characters provide line breaking opportunities and that UAs will not break at every opportunity, but only near the end of a line unless a line break is forced.
- [ ] The device is a full-color device.
- [ ] The device has a viewport width of at least 640px (approx).
- [ ] The resolution of the device is 96 CSS pixels per inch.
- [ ] The UA imposes no minimum font size.
- [x] The 'medium' font-size computes to 16px.
- [x] The initial value of 'color' is black.
- [x] The canvas background is white.
- [ ] The user stylesheet is empty (except where indicated by the tests).
- [ ] The device is interactive and uses scroll bars.

Other things we nowadays assume include:
- [ ] Ahem is installed. (We may want to go as far as checking we have an up-to-date version.)

### PR DESCRIPTION
**Pull request by gsnedders**
This starts to resolve #3827

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/4411)

### CODE REVIEW

File: assumptions/html-elements.html [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+}, "p is display: block");
+
+test(function() {
+  var ul_li = document.getElementsByTagName("li")[0];
+  var styles = window.getComputedStyle(ul_li);
+  assert_equals(styles["display"], "list-item");
+}, "ul > li is display: block");
+
+test(function() {
+  var ol_li = document.getElementsByTagName("li")[0];
```
Comment (Ms2ger) [inline assumptions/html-elements.html]: This one's `ul > li`, not `ol > li`

File: assumptions/html-elements.html [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  var b = document.querySelector("span.b");
+
+  var a_styles = window.getComputedStyle(a);
+  var b_styles = window.getComputedStyle(b);
+
+  assert_equals(a_styles.length, b_styles.length, "Same properties on both #a and #b");
+
+  for (var i = 0; i < a_styles.length; i++) {
+    var property = a_styles[i];
+    assert_equals(property, b_styles[i], "Same property on #a and #b");
```
Comment (Ms2ger) [inline assumptions/html-elements.html]: I'm not quite sure if all of this is guaranteed, but I'm not going to worry about it until someone complains.

**Review by Ms2ger — APPROVED** [review]
r+ if you fix the `li`.

File: assumptions/html-elements.html (hunk 2 above)
Comment (gsnedders) [inline assumptions/html-elements.html]: Not sure why?

Comment (gsnedders) [inline assumptions/html-elements.html]: Like, why aren't you sure if it's all guaranteed?

**Comment by tobie** [comment]
Pinging @wpt-pr-bot.

**Comment by w3c-bots** [comment]
# Chrome (unstable channel) #
Testing web-platform-tests at revision 479e16e56853392dd36d2bb3283704443790597c
Using browser at version 58.0.3004.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/4411/assumptions/initial-color.html">/assumptions/initial-color.html</a></summary>


| Subtest | Results |
|---------|---------|
|         | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4411/assumptions/min-font-size.html">/assumptions/min-font-size.html</a></summary>


| Subtest | Results |
|---------|---------|
|         | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4411/assumptions/html-elements.html">/assumptions/html-elements.html</a></summary>


|                            Subtest                            | Results |
|---------------------------------------------------------------|---------|
|                                                               | OK      |
| `table is display: table`                                     | PASS    |
| `(pre-req for comparison tests) initial CSS value supported`  | PASS    |
| `Compare CSS span definitions (only valid if pre-reqs pass)`  | PASS    |
| `ul > li is display: block`                                   | PASS    |
| `ol > li is display: block`                                   | PASS    |
| `tbody is display: table-row-group`                           | PASS    |
| `(pre-req for comparison tests) all CSS short-hand supported` | PASS    |
| `Compare CSS div definitions (only valid if pre-reqs pass)`   | PASS    |
| `p is display: block`                                         | PASS    |
| `tr is display: table-row`                                    | PASS    |
| `td is display: table-cell`                                   | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4411/assumptions/medium-font-size.html">/assumptions/medium-font-size.html</a></summary>


| Subtest | Results |
|---------|---------|
|         | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4411/assumptions/canvas-background.html">/assumptions/canvas-background.html</a></summary>


| Subtest | Results |
|---------|---------|
|         | PASS    |

</details>

**Comment by w3c-bots** [comment]
# Firefox (nightly channel) #
Testing web-platform-tests at revision 479e16e56853392dd36d2bb3283704443790597c
Using browser at version BuildID 20170123125947; SourceStamp 36486fdc3813ef7943ae5b07b4128866d1938a6c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/4411/assumptions/initial-color.html">/assumptions/initial-color.html</a></summary>


| Subtest | Results |
|---------|---------|
|         | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4411/assumptions/min-font-size.html">/assumptions/min-font-size.html</a></summary>


| Subtest | Results |
|---------|---------|
|         | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4411/assumptions/html-elements.html">/assumptions/html-elements.html</a></summary>


|                            Subtest                            | Results |
|---------------------------------------------------------------|---------|
|                                                               | OK      |
| `table is display: table`                                     | PASS    |
| `(pre-req for comparison tests) initial CSS value supported`  | FAIL    |
| `Compare CSS span definitions (only valid if pre-reqs pass)`  | PASS    |
| `ul > li is display: block`                                   | PASS    |
| `ol > li is display: block`                                   | PASS    |
| `tbody is display: table-row-group`                           | PASS    |
| `(pre-req for comparison tests) all CSS short-hand supported` | PASS    |
| `Compare CSS div definitions (only valid if pre-reqs pass)`   | FAIL    |
| `p is display: block`                                         | PASS    |
| `tr is display: table-row`                                    | PASS    |
| `td is display: table-cell`                                   | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4411/assumptions/medium-font-size.html">/assumptions/medium-font-size.html</a></summary>


| Subtest | Results |
|---------|---------|
|         | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4411/assumptions/canvas-background.html">/assumptions/canvas-background.html</a></summary>


| Subtest | Results |
|---------|---------|
|         | PASS    |

</details>

**Comment by Ms2ger** [comment]
w3c-test:mirror

**Comment by gsnedders** [comment]
@Ms2ger r?

File: assumptions/html-elements.html [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  </table>
+</div>
+<script>
+test(function() {
+  assert_true('all' in document.documentElement.style);
+}, "(pre-req for comparison tests) all CSS short-hand supported");
+
+test(function() {
+  assert_in_array(window.getComputedStyle(document.querySelector("div.c")).backgroundColor,
+                ["rgba(0, 0, 0, 0)", "transparent"]);
```
Comment (Ms2ger) [inline assumptions/html-elements.html]: Indentation

File: assumptions/html-elements.html [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+}, "(pre-req for comparison tests) initial CSS value supported");
+
+test(function() {
+  var a = document.querySelector("div.a");
+  var b = document.querySelector("div.b");
+
+  var a_styles = window.getComputedStyle(a);
+  var b_styles = window.getComputedStyle(b);
+
+  assert_equals(a_styles.length, b_styles.length, "Same properties on both #a and #b");
```
Comment (Ms2ger) [inline assumptions/html-elements.html]: Going to fix the `#a` references?

File: assumptions/html-elements.html [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  var p = document.getElementsByTagName("p")[0];
+  var styles = window.getComputedStyle(p);
+  assert_equals(styles["display"], "block");
+}, "p is display: block");
+
+test(function() {
+  var ul_li = document.querySelector("ul > li");
+  var styles = window.getComputedStyle(ul_li);
+  assert_equals(styles["display"], "list-item");
+}, "ul > li is display: block");
```
Comment (Ms2ger) [inline assumptions/html-elements.html]: No it isn't

File: assumptions/html-elements.html [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+}, "tbody is display: table-row-group");
+
+test(function() {
+  var tr = document.getElementsByTagName("tr")[0];
+  var styles = window.getComputedStyle(tr);
+  assert_equals(styles["display"], "table-row");
+}, "tr is display: table-row");
+
+test(function() {
+  var td = document.getElementsByTagName("td")[0];
```
Comment (Ms2ger) [inline assumptions/html-elements.html]: Worth testing `th`?

File: assumptions/min-font-size.html [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<!doctype html>
+<title>Check 1px and 2px font-sizes are different</title>
+<link rel="mismatch" href="min-font-size-ref.html">
```
Comment (Ms2ger) [inline assumptions/min-font-size.html]: Could use `-notref`. Not sure what our policy is on that.

**Review by Ms2ger — CHANGES_REQUESTED** [review]

File: assumptions/html-elements.html (hunk 4 above)
Comment (gsnedders) [inline assumptions/html-elements.html]: I thought I'd pushed that fix.

File: assumptions/min-font-size.html (hunk 7 above)
Comment (gsnedders) [inline assumptions/min-font-size.html]: There isn't one.

**Pull request merged**